frontend/src/metabase-types/api/search.ts

```diff
@@ -40,10 +40,23 @@ export interface SearchScore {
   column?: string;
 }
 
-export interface SearchResults {
-  data: SearchResult[];
-  models: SearchModelType[] | null;
-  available_models: SearchModelType[];
+interface BaseSearchResult<
+  Id extends SearchResultId,
+  Model extends SearchModelType,
+> {
+  id: Id;
+  model: Model;
+  name: string;
+}
+
+export interface SearchResults<
+  Id extends SearchResultId = SearchResultId,
+  Model extends SearchModelType = SearchModelType,
+  Result extends BaseSearchResult<Id, Model> = SearchResult<Id, Model>,
+> {
+  data: Result[];
+  models: Model[] | null;
+  available_models: Model[];
```

Comment:
I think that's not correct. `available_models` would get the list of all models that satisfy the search query, even if they were not set in `models`.

Reply:
Absolutely, thanks!
9586424f016c294e2007465c4ee21e7ddd886afc